src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -4127,7 +4127,7 @@ void validateSignature()
                 // Dynamic built-in operators allow virtually all operations with all types.  So we do no further checking here.
                 if (csharpReturnType.TypeKind is TypeKind.Dynamic ||
                     csharpLeftType.TypeKind is TypeKind.Dynamic ||
-                    csharpReturnType.TypeKind is TypeKind.Dynamic)
+                    csharpRightType.TypeKind is TypeKind.Dynamic)
```

Comment:
definitely looks like a good bug you've caught. :)

would need a test to go along with this.  Thsi also might be a breaking change in some way.  Compiler will have to look into that and see if this needs to be doc'ed.

Reply:
I don't think it will functionally change something because I can't seem to figure out how to have the right dynamic type without a return type that is also dynamic.